openedx/core/lib/block_structure/transformers.py

```diff
@@ -0,0 +1,100 @@
+"""
+Module for a collection of BlockStructureTransformers.
+"""
+from logging import getLogger
+
+from .exceptions import TransformerException
+from .transformer_registry import TransformerRegistry
+
+
+logger = getLogger(__name__)  # pylint: disable=C0103
+
+
+class BlockStructureTransformers(object):
+    """
+    Class for a collection of block structure transformers.
+    """
+    def __init__(self, transformers=None, usage_info=None):
+        """
+        Arguments:
+            transformers ([BlockStructureTransformer]) - List of transformers
+                to add to the collection.
+
+            usage_info (any negotiated type) - A usage-specific object
```

Comment:
Yeah, this question came up before as well (in the initial design).  I'm still not sure how `usage_info` will evolve.  For now, if we continue to keep this layer course-and-user-agnostic, then the `usage_info` needs to be injected down and forwarded to the transformers.

I was envisioning different higher layers would have different `usage_info` needs for their own set of transformers and so it would be passed in.  I'm open to suggestions on how to improve this model.

This PR doesn't change the usage/design of the `usage_info`, so we can also improve it outside of this PR.